ngraph/core/reference/src/runtime/reference/fft.cpp

```diff
@@ -0,0 +1,594 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/reference/fft.hpp"
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <complex>
+#include <cstring>
+#include <functional>
+#include <utility>
+#include <vector>
+#include "ngraph/shape.hpp"
+
+using namespace ngraph;
+using namespace ngraph::runtime::reference;
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            std::vector<int64_t> canonicalize_axes(const int64_t* axes_data,
```

Comment:
Can you add more comments in the code? For example I don't understand (maybe because I don't know the algorithm) why do we need to have this method

Reply:
Done.